src/main/kotlin/app/revanced/patches/youtubevanced/ad/general/patch/GeneralAdsRemovalPatch.kt

```diff
@@ -0,0 +1,64 @@
+package app.revanced.patches.youtubevanced.ad.general.patch
+
+import app.revanced.extensions.toErrorResult
+import app.revanced.patcher.annotation.Description
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.BytecodeContext
+import app.revanced.patcher.extensions.addInstructions
+import app.revanced.patcher.patch.BytecodePatch
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.annotations.DependsOn
+import app.revanced.patcher.patch.annotations.Patch
+import app.revanced.patches.youtube.misc.fix.backtoexitgesture.patch.FixBackToExitGesturePatch
+import app.revanced.patches.youtube.misc.fix.verticalscroll.patch.VerticalScrollPatch
+import app.revanced.patches.youtubevanced.ad.general.annotations.YouTubeVancedCompatibility
+import app.revanced.patches.youtubevanced.ad.general.fingerprints.ContainsAdFingerprintPrimary
+import app.revanced.patches.youtubevanced.ad.general.fingerprints.ContainsAdFingerprintSecondary
+import org.jf.dexlib2.iface.instruction.formats.Instruction21c
+
+@Patch
+@Name("general-ad-removal")
+@Description("Removes general ads from YouTube Vanced.")
+@DependsOn([VerticalScrollPatch::class])
+@YouTubeVancedCompatibility
+@Version("0.0.1")
+class GeneralAdsRemovalPatch : BytecodePatch(
+    listOf(
+        ContainsAdFingerprintPrimary,
+        ContainsAdFingerprintSecondary
+    )
+) {
+    override fun execute(context: BytecodeContext): PatchResult {
+        listOf(
+            ContainsAdFingerprintPrimary,
+            ContainsAdFingerprintSecondary
+        ).forEach {
+            val result = it.result ?: return it.toErrorResult()
+            val insertIndex = result.scanResult.patternScanResult!!.endIndex + 1
+            with(result.mutableMethod) {
+                val register = (implementation!!.instructions[insertIndex - 2] as Instruction21c).registerA
```

Comment:
Don't forget to import the extension function:
```suggestion
                val register = (instruction(insertIndex - 2) as Instruction21c).registerA
```